rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.pnc.rest.notifications.model;
+
+import org.jboss.pnc.spi.BuildStatus;
+import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.Map;
+
+@ApplicationScoped
+public class NotificationFactory {
+
+    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
+
+    public NotificationFactory() {
+        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
```

Comment:
The signal event name (for bpm) is intentionally passed in URL to avoid storing it in 2 places (PNC & BPM server). Thus, the entiner callback url construction is up to BPM server and PNC just tries to invoke what has been passed to it. If you believe it should be refactor, then please go ahead, I'm not against it.

In either way, this code will need to be amended at some point because we might need to report only final state of the build to BPM server. Currently, it reports other events, like repo creation If I'm not mistaken.